src/pages/index.js

```diff
@@ -54,43 +58,246 @@ const addFormValidator = new FormValidator(
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .updateUserInfo(data["edit-profile"], data["profile-description"])
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.classList.remove("loading");
+      saveButton.disabled = false;
+    });
 }
 
 function handleAddCardFormSubmit(data) {
-  const newCardElement = createCard({ name: data.title, link: data.url });
-  section.addItem(newCardElement);
-  addCardFormElement.reset();
-  addFormValidator.disableButton();
-  addCardPopup.close();
+  const sanitizedUrl = data.url.replace(/["'\s]+$/, "").trim();
+  const saveButton = addCardFormElement.querySelector(".modal__save-button");
+  saveButton.textContent = "Saving...";
+  saveButton.classList.add("loading");
+  saveButton.disabled = true;
+  api
+    .addCard(data.title, sanitizedUrl)
+    .then((newCardInfo) => {
+      const newCardElement = createCard({
+        name: newCardInfo.name,
+        link: newCardInfo.link,
+        _id: newCardInfo._id,
+      });
+      section.addItem(newCardElement);
+      addCardFormElement.reset();
+      addFormValidator.resetValidation();
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error adding new card:", err);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      saveButton.classList.remove("loading");
+      saveButton.disabled = false;
+      addCardPopup.close();
+    });
 }
 
+function handleDeleteConfirm(data) {
+  const cardId = data.cardId;
+  if (!cardId || typeof cardId !== "string") {
+    console.error(
+      "Invalid cardId in handleDeleteConfirm:",
+      cardId,
+      "Data:",
+      data
+    );
+    deleteCardPopup.close();
+    return;
+  }
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      section.removeItem(cardId);
+      deleteCardPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+      deleteCardPopup.close();
+    });
+}
+
+const likeStatuses = new Map();
+
+function handleLikeClick(cardId) {
+  const currentIsLiked = likeStatuses.get(cardId) || false;
+  api
+    .toggleLike(cardId, currentIsLiked)
+    .then((updatedCard) => {
+      likeStatuses.set(cardId, updatedCard.isLiked);
+      const cardElement = document.querySelector(`.card[data-id="${cardId}"]`);
+      if (cardElement) {
+        const likeButton = cardElement.querySelector(".card__like-button");
+        if (likeButton) {
+          likeButton.classList.toggle(
+            "card__like-button_active",
+            updatedCard.isLiked
+          );
```

Comment:
* Now only the `Card` class knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle